integrations/pinecone/src/pinecone_haystack/filters.py

```diff
@@ -0,0 +1,193 @@
+    supported_types = (int, float)
+    if not isinstance(value, supported_types):
+        msg = (
+            f"Unsupported type for 'greater than' comparison: {type(value)}. "
+            f"Types supported by Pinecone are: {supported_types}"
+        )
+        raise FilterError(msg)
+
+    return {field: {"$gt": value}}
+
+
+def _greater_than_equal(field: str, value: Any) -> Dict[str, Any]:
+    supported_types = (int, float)
+    if not isinstance(value, supported_types):
+        msg = (
+            f"Unsupported type for 'greater than equal' comparison: {type(value)}. "
+            f"Types supported by Pinecone are: {supported_types}"
+        )
+        raise FilterError(msg)
+
+    return {field: {"$gte": value}}
+
+
+def _less_than(field: str, value: Any) -> Dict[str, Any]:
+    supported_types = (int, float)
+    if not isinstance(value, supported_types):
+        msg = (
+            f"Unsupported type for 'less than' comparison: {type(value)}. "
+            f"Types supported by Pinecone are: {supported_types}"
+        )
+        raise FilterError(msg)
+
+    return {field: {"$lt": value}}
+
+
+def _less_than_equal(field: str, value: Any) -> Dict[str, Any]:
+    supported_types = (int, float)
+    if not isinstance(value, supported_types):
+        msg = (
+            f"Unsupported type for 'less than equal' comparison: {type(value)}. "
+            f"Types supported by Pinecone are: {supported_types}"
+        )
+        raise FilterError(msg)
+
+    return {field: {"$lte": value}}
+
+
+def _not_in(field: str, value: Any) -> Dict[str, Any]:
+    if not isinstance(value, list):
+        msg = f"{field}'s value must be a list when using 'not in' comparator in Pinecone"
+        raise FilterError(msg)
+
+    supported_types = (int, float, str)
+    for v in value:
+        if not isinstance(v, supported_types):
+            msg = (
+                f"Unsupported type for 'not in' comparison: {type(v)}. "
+                f"Types supported by Pinecone are: {supported_types}"
+            )
+            raise FilterError(msg)
+
+    return {field: {"$nin": value}}
+
+
+def _in(field: str, value: Any) -> Dict[str, Any]:
+    if not isinstance(value, list):
+        msg = f"{field}'s value must be a list when using 'in' comparator in Pinecone"
+        raise FilterError(msg)
+
+    supported_types = (int, float, str)
+    for v in value:
+        if not isinstance(v, supported_types):
+            msg = (
+                f"Unsupported type for 'in' comparison: {type(v)}. "
+                f"Types supported by Pinecone are: {supported_types}"
+            )
+            raise FilterError(msg)
+
+    return {field: {"$in": value}}
+
+
+COMPARISON_OPERATORS = {
+    "==": _equal,
+    "!=": _not_equal,
+    ">": _greater_than,
+    ">=": _greater_than_equal,
+    "<": _less_than,
+    "<=": _less_than_equal,
+    "in": _in,
+    "not in": _not_in,
+}
+
+LOGICAL_OPERATORS = {"AND": "$and", "OR": "$or"}
```

Comment:
I would put module constants at the top

Reply:
The fact is that `COMPARISON_OPERATORS` depends on some functions, so it should live after the function definitions.
And I would prefer `LOGICAL_OPERATORS` to be close.